hack/generator/pkg/astmodel/property_definition.go

```diff
@@ -22,7 +22,7 @@ type PropertyDefinition struct {
 	propertyType Type
 	description  string
 	validations  []Validation
-	tags         map[string][]string
+	tags         map[string][]string // TODO: it might be easier if this were a map[string]map[string]struct{}?
```

Comment:
I'd say do the change, it will make the removal code a lot simpler.

Reply:
The downside is we lose control over ordering of the tags, which could make them look a little wonky.

Anyway, will do this for now and we can deal with the ordering issue later if we still don't like it...

Reply:
Actually that downside is quite crippling:
https://play.golang.org/p/pqD10SfnRdq

Order matters for the JSON tags, so I think it needs to be an array.